File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/mad/Java/Go/s826491904_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	sc := bufio.NewScanner(os.Stdin)
	sc.Scan()
	N, _ := strconv.Atoi(sc.Text())

	sc.Scan()
	max, _ := strconv.Atoi(sc.Text())
	min := max

	for i := 1; i < N; i++ {
		sc.Scan()
		in, _ := strconv.Atoi(sc.Text())
		if in > max {
			max = in
		}
		if in < min {
			min = in
		}
	}
	fmt.Println(max - min)
}

// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/comment/uncomment/direct/Java/C/s826491904_commented.c
#include <stdio.h>

int main() {
    int N;
    scanf("%d", &N);
    int max, min;
    int temp;
    scanf("%d", &temp);
    max = min = temp;
    int in;
    for (int i = 1; i < N; i++) {
        scanf("%d", &in);
        if (in > max) max = in;
        if (in < min) min = in;
    }
    printf("%d\n", max - min);
    return 0;
}
// End of Code
